packages/@react-spectrum/s2/src/Tabs.tsx

```diff
@@ -10,9 +10,320 @@
  * governing permissions and limitations under the License.
  */
 
-import {Tabs as RACTabs, TabsProps} from 'react-aria-components';
+import {TabListProps as AriaTabListProps, TabPanel as AriaTabPanel, TabPanelProps as AriaTabPanelProps, TabProps as AriaTabProps, TabsProps as AriaTabsProps, Provider, Tab as RACTab, TabList as RACTabList, Tabs as RACTabs, TabListStateContext} from 'react-aria-components';
+import {centerBaseline} from './CenterBaseline';
+import {Collection, DOMRef, Key, Node, Orientation} from '@react-types/shared';
+import {createContext, forwardRef, ReactNode, useCallback, useContext, useEffect, useRef, useState} from 'react';
+import {focusRing, getAllowedOverrides, UnsafeStyles, StylesPropWithHeight, StyleProps} from './style-utils' with {type: 'macro'};
+import {IconContext} from './Icon';
+import {mergeStyles} from '../style/runtime';
+import {size, style} from '../style/spectrum-theme' with {type: 'macro'};
+import {Text, TextContext} from './Content';
+import {useDOMRef} from '@react-spectrum/utils';
+import {useLayoutEffect} from '@react-aria/utils';
+import {useLocale} from '@react-aria/i18n';
 
+interface TabsProps extends Omit<AriaTabsProps, 'className' | 'style' | 'children'>, UnsafeStyles {
+  /** Spectrum-defined styles, returned by the `style()` macro. */
+  styles?: StylesPropWithHeight,
+  /** The content to display in the tabs. */
+  children?: ReactNode,
+  /** The amount of space between the tabs. */
+  density?: 'compact' | 'regular'
+}
+
+interface TabProps extends Omit<AriaTabProps, 'children' | 'style' | 'className'>, StyleProps {
+  children?: ReactNode
+}
+
+interface TabListProps<T> extends Omit<AriaTabListProps<T>, 'children' | 'style' | 'className'>, StyleProps {
+  children?: ReactNode
+}
+
+interface TabPanelProps extends Omit<AriaTabPanelProps, 'children' | 'style' | 'className'>, StyleProps {
+  children?: ReactNode
+}
+
+export function TabPanel(props: TabPanelProps) {
+  return (
+    <AriaTabPanel 
+      {...props}
+      style={props.UNSAFE_style}
+      className={(props.UNSAFE_className || '') + style({marginTop: 4, color: 'gray-800'}, getAllowedOverrides())(null, props.styles)} />
+  );
+}
+
+const tab = style({
```

Comment:
when a tab is icon-only, it appears there should be some padding in the X direction, I noticed that the selection indicator sticks out over the sides. I don't see a token for it, so you may need to ask for that

Reply:
![Screenshot 2024-08-07 at 12 06 27 PM](https://github.com/user-attachments/assets/83451cba-958c-4620-b3fa-d3d8b2089966)

Reply:
interesting, yeah i definitely looks like there should be padding. the tokens seem to say otherwise cause there's a section under `Spacing (edge to edge)` (not sure if i want to post a pic of the tokens here) where it's defined as none but that might only be for tabs that aren't icon only so i've reached out to the spectrum team for clarification